src/main/scala/org/tygus/suslik/synthesis/Synthesis.scala

```diff
@@ -50,7 +51,33 @@ trait Synthesis extends SepLogicUtils {
 
   }
 
-  private def synthesize(goal: Goal, depth: Int)
+  var saved_results = scala.collection.mutable.Map[Goal, (Option[Statement], Int)]()
```

Comment:
can you remove memoization from this PR? I don't think we want to merge it because the performance benefits are inconsistent as Ilya pointed out (some benchmarks get slower), and it complicates the code